coralibre-sdk/sdk/src/main/java/org/coralibre/android/sdk/internal/crypto/ppcp/CryptoModule.java

```diff
@@ -33,54 +29,78 @@
     public static final String AEMK_INFO = "EN-AEMK";
 
     private static CryptoModule instance;
-
-    private SharedPreferences esp;
     private ENNumber currentTekDay = new ENNumber(0);
     private RollingProximityIdentifierKey currentRPIK;
     private AssociatedEncryptedMetadataKey currentAEMK;
     private BluetoothPayload currentPayload = null;
     private AssociatedMetadata metadata = null;
+    private Database database;
 
-    public static CryptoModule getInstance(Context context) {
-        try {
-            if (instance == null) {
-                instance = new CryptoModule();
-                String KEY_ALIAS = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC);
-                instance.esp = EncryptedSharedPreferences.create("coralibre_store",
-                        KEY_ALIAS,
-                        context,
-                        EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
-                        EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM);
-
-                //TODO: use dependency injection
-                Database database = MockDatabase.getInstance();
-                GeneratedTEK rawTek = database.getGeneratedTEK(
-                        TemporaryExposureKey.getMidnight(getCurrentInterval()));
-                if(rawTek == null) {
-                    instance.updateTEK();
-                } else {
-                    TemporaryExposureKey tek = new TemporaryExposureKey(rawTek.getInterval(), rawTek.getKey());
-                    instance.currentTekDay = tek.getInterval();
-                    instance.currentRPIK = generateRPIK(tek);
-                    instance.currentAEMK = generateAEMK(tek);
-                }
+    private boolean testMode;
+    private ENNumber currentInterval;
+
+    public CryptoModule() {
+        //TODO: Use propper dependency injection
+        init(new MockDatabase(), getCurrentInterval());
+    }
 
+    // If you call this method your warranty will be void.
```

Comment:
What does this comment mean? ;-)

Reply:
I added a description for why one should not use this Constouctor, however it was good to have something like a
```
@waring("testing only")
```
in front of this constructor. I don't know if something like this exists in java.

Reply:
Maybe making it private and in tests calling it via reflection would be a viable solution. Or putting a `@Deprecated` annotation along with an explanation of why it is deprecated in the `javadoc`. Anyway, I do not like the idea of having test-only code in the main sdk. But maybe fixing this should be reconsidered when tests will be implemented, for now the comment is enough. Could you convert it to a `javadoc` comment though, so that it can be viewed via IDEs?

Reply:
Its not possible to inject a mocked time into the code without changing the sdk code. And custom time is viable for testing. Otherwise testing was not really possible.

Reply:
done

Reply:
Yeah ok